vscode-extension/src/extension.ts

```diff
@@ -1,27 +1,290 @@
 // The module 'vscode' contains the VS Code extensibility API
 // Import the module and reference it with the alias vscode in your code below
-import * as vscode from 'vscode';
+import { workspace, window, commands, extensions, Uri, ExtensionContext } from 'vscode';
+import * as process from 'process';
+import * as cp from 'child_process';
+import * as util from 'util';
+import { posix } from 'path';
 
 // This method is called when your extension is activated
 // Your extension is activated the very first time the command is executed
-export function activate(context: vscode.ExtensionContext) {
-
-	// Use the console to output diagnostic information (console.log) and errors (console.error)
+export function activate(context: ExtensionContext) {
 	// This line of code will only be executed once when your extension is activated
-	vscode.window.onDidOpenTerminal(async (event) => {
-		runDevboxShell();
+	initialCheckDevboxJSON();
+	// Creating file watchers to watch for events on devbox.json
+	const fswatcher = workspace.createFileSystemWatcher("**/devbox.json", false, false, false);
+	fswatcher.onDidDelete(e => {
+		commands.executeCommand('setContext', 'devbox.configFileExists', false);
+	});
+	fswatcher.onDidCreate(e => {
+		commands.executeCommand('setContext', 'devbox.configFileExists', true);
+	});
+	fswatcher.onDidChange(e => {
+		initialCheckDevboxJSON();
+	});
+
+	// Check for devbox.json when a new folder is opened
+	workspace.onDidChangeWorkspaceFolders(async (e) => {
+		initialCheckDevboxJSON();
+	});
+
+	// run devbox shell when terminal is opened
+	window.onDidOpenTerminal(async (e) => {
+		if (workspace.getConfiguration("devbox").get("autoShellOnTerminal")) {
+			runDevboxShell();
+		}
+	});
+
+	const setupDevcontainer = commands.registerCommand('devbox.setupDevContainer', async () => {
+		const exec = util.promisify(cp.exec);
+		// determining cpu architecture - needed for devcontainer dockerfile
+		const { stdout, stderr } = await exec("uname -m");
+		let cpuArch = stdout;
+		if (stderr) {
+			console.log(stderr);
+			const response = await window.showErrorMessage(
+				"Could not determine the CPU architecture type. Is your architecture type Apple M1/arm64?",
+				"Yes",
+				"No",
+			);
+			cpuArch = response === "Yes" ? "arm64" : "something else";
```

Comment:
something else?

Reply:
the value of "something else doesn't matter as long as it's not `arm64` but I change the phrase to `"undefined"`